### ISSUE TITLE
Testing on PASCAL VOC is very slow

### ISSUE THREAD

**Issue by PardoAlejo** [NVlabs/GroupViT#16]
Hi! 
Currently, I'm working with your code and have been able to train in multiple GPUs without major issues. However, I had to disable the testing on PASCAL VOC because it takes forever and the code gets stuck. the last prints that I can see are the next ones:
```
All checkpoints founded in output/group_vit_gcc_3m_bs712x4: []
All checkpoints founded in output/group_vit_gcc_3m_bs712x4: []
All checkpoints founded in output/group_vit_gcc_3m_bs712x4: []
All checkpoints founded in output/group_vit_gcc_3m_bs712x4: []
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0


[                           ] 20/1449, 0.0 task/s, elapsed: 9609s, ETA: 686561s(base)
```
And
```
[2022-04-10 10:10:18 group_vit_gcc_3m_bs712x4](main_group_vit.py 294): INFO EPOCH 0 training takes 0:27:05
[2022-04-10 10:10:19 group_vit_gcc_3m_bs712x4](checkpoint.py 92): INFO output/group_vit_gcc_3m_bs712x4/ckpt_epoch_0.pth saving......
[2022-04-10 10:10:20 group_vit_gcc_3m_bs712x4](checkpoint.py 95): INFO output/group_vit_gcc_3m_bs712x4/ckpt_epoch_0.pth saved !!!
[2022-04-10 10:10:20 group_vit_gcc_3m_bs712x4](main_group_vit.py 160): INFO Avg loss of the network on the 2876225 train images: 15.16
[2022-04-10 10:10:20 group_vit_gcc_3m_bs712x4](main_group_vit.py 317): INFO Building zero shot classifier
[2022-04-10 10:10:25 group_vit_gcc_3m_bs712x4](main_group_vit.py 321): INFO Zero shot classifier built
[2022-04-10 10:10:39 group_vit_gcc_3m_bs712x4](main_group_vit.py 347): INFO Test: [0/17]        Time 19.063 (19.063)    Loss 6.8310 (6.8310)    Acc@1 1.545 (1.545)     Acc@5 6.180 (6.180)     Mem 71248MB
[2022-04-10 10:12:43 group_vit_gcc_3m_bs712x4](main_group_vit.py 347): INFO Test: [10/17]       Time 13.185 (13.037)    Loss 6.8311 (6.8282)    Acc@1 1.990 (1.904)     Acc@5 6.739 (6.734)     Mem 71248MB
[2022-04-10 10:13:53 group_vit_gcc_3m_bs712x4](main_group_vit.py 353): INFO Clearing zero shot classifier
[2022-04-10 10:13:53 group_vit_gcc_3m_bs712x4](main_group_vit.py 355): INFO  * Acc@1 1.903 Acc@5 6.735
[2022-04-10 10:13:53 group_vit_gcc_3m_bs712x4](main_group_vit.py 166): INFO Accuracy of the network on the 50000 test images: 1.9%
[2022-04-10 10:13:53 group_vit_gcc_3m_bs712x4](checkpoint.py 92): INFO output/group_vit_gcc_3m_bs712x4/ckpt_epoch_0_best_acc1.pth saving......
[2022-04-10 10:13:55 group_vit_gcc_3m_bs712x4](checkpoint.py 95): INFO output/group_vit_gcc_3m_bs712x4/ckpt_epoch_0_best_acc1.pth saved !!!
[2022-04-10 10:13:55 group_vit_gcc_3m_bs712x4](main_group_vit.py 173): INFO Max accuracy: 1.90%
[2022-04-10 10:13:55 group_vit_gcc_3m_bs712x4](group_vit_seg.py 142): INFO Building GroupViTSegInference with 21 classes, test_cfg=Config (path: None): {'bg_thresh': 0.95, 'mode': 'slide', 'stride': (224, 224), 'crop_size': (448, 448)}, with_bg=True
```

Do you have any clue about what is going on? Thanks for your help!

**Comment by xvjiarui** [comment]
Hi @PardoAlejo 
May I know which command you ran? 
Did you encounter similar issue when running test with provided model?

**Comment by PardoAlejo** [comment]
Hi @xvjiarui thanks for replying. 
The logs that I attached above was using the training command suggested in the README.me.
However, I also tried with the testing command but with a model trained by me on CC3M. 
The outcome for both cases was the same, the code gets stuck there and the ETA is unreasonable.

**Comment by xvjiarui** [comment]
Did you try `./tools/dist_launch.sh main_seg.py /path/to/config 8 --resume https://github.com/xvjiarui/GroupViT-1/releases/download/v1.0.0/group_vit_gcc_yfcc_30e-879422e0.pth`?

**Comment by PardoAlejo** [comment]
I just tried and the outcome is the same. The only difference is that I ran it with 4 GPUs not with 8 as you suggested. The time reduces but still take a lot, the ETA is around 24 hours:
```
Downloading: "https://github.com/xvjiarui/GroupViT-1/releases/download/v1.0.0/group_vit_gcc_yfcc_30e-879422e0.pth" to /home/pardogl/.cache/torch/hub/checkpoints/group_vit_gcc_yfcc_30e-879422e0.pth
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 213M/213M [00:08<00:00, 26.1MB/s]
[2022-04-11 09:51:47 group_vit_gcc_3m_bs512x4](checkpoint.py 51): INFO <All keys matched successfully>
[2022-04-11 09:51:47 group_vit_gcc_3m_bs512x4](main_group_vit.py 372): INFO
[2022-04-11 09:51:50 group_vit_gcc_3m_bs512x4](group_vit_seg.py 142): INFO Building GroupViTSegInference with 21 classes, test_cfg=Config (path: None): {'bg_thresh': 0.95, 'mode': 'slide', 'stride': (224, 224), 'crop_size': (448, 448)}, with_bg=True
[2022-04-11 09:51:50 group_vit_gcc_3m_bs512x4](main_group_vit.py 374): INFO Seg Model Initialized
[2022-04-11 09:51:50 group_vit_gcc_3m_bs512x4](main_group_vit.py 378): INFO Model set to eval
[                                                  ] 4/1449, 0.0 task/s, elapsed: 234s, ETA: 84358s
```
Is it supposed to be that slow?

**Comment by xvjiarui** [comment]
That's strange. 
I tried with my RTX3090. 
On single GPU: 
./tools/dist_launch.sh main_seg.py configs/group_vit_gcc_yfcc_30e.yml 1 --resume https://github.com/xvjiarui/GroupViT-1/releases/download/v1.0.0/group_vit_gcc_yfcc_30e-879422e0.pth
[2022-04-11 00:42:02 group_vit_gcc_yfcc_30e_bs256x1](group_vit_seg.py 142): INFO Building GroupViTSegInference with 21 classes, test_cfg=Config (path: None): {'bg_thresh': 0.95, 'mode': 'slide', 'stride': (224, 224), 'crop_size': (448, 448)}, with_bg=True
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 1449/1449, 9.9 task/s, elapsed: 147s, ETA:     0sper class results:
One 2 GPUs: 
./tools/dist_launch.sh main_seg.py configs/group_vit_gcc_yfcc_30e.yml 2 --resume https://github.com/xvjiarui/GroupViT-1/releases/download/v1.0.0/group_vit_gcc_yfcc_30e-879422e0.pth
[2022-04-11 00:40:16 group_vit_gcc_yfcc_30e_bs256x2](group_vit_seg.py 142): INFO Building GroupViTSegInference with 21 classes, test_cfg=Config (path: None): {'bg_thresh': 0.95, 'mode': 'slide', 'stride': (224, 224), 'crop_size': (448, 448)}, with_bg=True
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 1450/1449, 21.7 task/s, elapsed: 67s, ETA:     0sper class results:


I am using PyTorch 1.8. 
Did you happen to modify the code?

**Comment by PardoAlejo** [comment]
I actually did, thank you! I was plotting the attention maps after every grouping block, that's why it was so slow. My bad. Thank you very much for your help!

**Comment by pzhren** [comment]
(1) During pre-training, I found that the inference speed on imagenet is very slow (fast on VOC), almost stuck, why is this happening? The following log information is also very strange, why is acc greater than 100?
`2022-05-09 00:31:50 group_vit_gcc12m_30e_bs256x8](main_group_vit.py 347):INFO Test:[0/24]
Time16.943(16.943)
Loss   74103264063228363208700061926031360.0000 (74103264063228363208700061926031360.0000)
Acc@1  6037924414820376253962026343202816.000  (6037924414820376253962026343202816.000)
Acc@5   27701119339920547340038874347864064.000 (27701119339920547340038874347864064.000) Mem 26158MB`
(2) Is the inference weight on the coco, voc and other data sets the best weight for the verification result on the imagenet used?

**Comment by xvjiarui** [comment]
Hi @pzhren 
1. Did you try to run inference with the provided weight?
2. Yes. They are the same.

**Comment by pzhren** [comment]
> Hi @pzhren
> 
> 1. Did you try to run inference with the provided weight?
> 2. Yes. They are the same.
I tried the provided weights and imagenet is really slow, almost stuck.
I did pre-training on gcc12m, but the result is poor. I am guessing if there is a problem with my dataset download, I would like to know how big is the folder size of your gcc12m_shards?

**Comment by pzhren** [comment]
Hi @xvjiarui 
I tried the provided weights and imagenet is really slow, almost stuck.
I did pre-training on gcc12m, but the result is poor. I am guessing if there is a problem with my dataset download, I would like to know how big is the folder size of your gcc12m_shards?

**Comment by xvjiarui** [comment]
Hi @pzhren 

The inference on ImageNet should be finished in 2 mins. 
My `gcc12m_shards` is of size 1.8T.

**Comment by pzhren** [comment]
Hi @xvjiarui 
Thank you very much. I am sure that it is the problem of my gcc12m data set. It only has 20G. It is very strange that it can be trained without any error. I don't know why this is. Is your gcc3m around 200G? Are there VOC results on gcc3m?

**Comment by xvjiarui** [comment]
Hi @pzhren 
I don't have results on gcc3m since it's too small for image text contrastive. 
My gcc3m is around 414G. 

Did you successfully reproduce the inference results with ImageNet?

**Comment by pzhren** [comment]
Hi @xvjiarui 
No. I use img2dataset to download the dataset, gcc3m is only about 200G, but it can also train normally and no error is reported, but the result is very poor, the miou of voc is less than 10. There are some gcc data sets locally, but they are not in webdataset format. I have to go to a data interface to run them.

**Comment by pzhren** [comment]
> hi, are there any pre-trained weights on gcc12m+yfcc14m to share?

**Comment by xvjiarui** [comment]
Hi @pzhren 

Please find checkpoints [here](https://github.com/xvjiarui/GroupViT#benchmark-results)

**Comment by pzhren** [comment]
hi, is there a weight of GCC12M+YFCC14M? I need to compare with this setup, thanks you




------------------&nbsp;Original&nbsp;------------------
From: "Yu ***@***.***&gt;; 
Date: 2022年9月19日(星期一) 晚上10:41
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: Re: [NVlabs/GroupViT] Testing on PASCAL VOC is very slow (Issue #16)





 
Hi @pzhren
 
Please find checkpoints here

**Comment by xvjiarui** [comment]
The first weight is GCC12m+YFCC14M

**Comment by pzhren** [comment]
Haha, the weight group_vit_gcc_yfcc_30e-879422e0.pth is the weight of GCC3M+GCC12m+YFCC14M, right?




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "NVlabs/GroupViT"                                                                                    ***@***.***&gt;;
Date:&nbsp;Mon, Sep 19, 2022 11:26 PM
***@***.***&gt;;
***@***.******@***.***&gt;;
Subject:&nbsp;Re: [NVlabs/GroupViT] Testing on PASCAL VOC is very slow (Issue #16)





 
The first weight is GCC12m+YFCC14M

**Comment by xvjiarui** [comment]
> Haha, the weight group_vit_gcc_yfcc_30e-879422e0.pth is the weight of GCC3M+GCC12m+YFCC14M, right?
> […](#)

Yes. That's correct.
